pkg/apis/core/v1/defaults.go

```diff
@@ -176,6 +176,36 @@ func SetDefaults_Pod(obj *v1.Pod) {
 				}
 			}
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			// set resources allocated equal to requests, if not specified
+			if obj.Spec.Containers[i].Resources.Requests != nil {
+				if obj.Spec.Containers[i].ResourcesAllocated == nil {
+					obj.Spec.Containers[i].ResourcesAllocated = make(v1.ResourceList)
+					for key, value := range obj.Spec.Containers[i].Resources.Requests {
+						obj.Spec.Containers[i].ResourcesAllocated[key] = value.DeepCopy()
+					}
+				}
+			}
+			// set resize policy to defaults, if not specified
+			resources := make(map[v1.ResourceName]bool)
+			for _, p := range obj.Spec.Containers[i].ResizePolicy {
+				resources[p.ResourceName] = true
+			}
+			if _, found := resources[v1.ResourceCPU]; !found {
```

Comment:
Comment why this is coded to support CPU and memory and nothing else?  What happens if some OTHER resource is provided (e.g. GPUs).  It doesn't get listed in resizePolicy?

Reply:
We currently plan to support in-place resize only for CPU and memory resources. So I restrict setting the policy only for cpu, memory in the admission plugin. But good point, maybe I should default all of them to NoRestart - we would have done that anyways if we did support other resource types. Let me think about this.

Reply:
The current implementation defaults both CPU and memory to NoRestart.